tools/stop_machine_aws.py

```diff
@@ -4,40 +4,28 @@
 import sys
 import logging
 import argparse
+import datetime
 
-def stop(instance, verbose):
+def stop(instance, logger):
     if instance.state['Name'] != 'stopped':
-            if verbose:
-                logger.info("Id: {0}, Platform: {1},Type: {2}, Public IPv4: {3}, AMI: {4}, State: {5}\n".format(instance.id, instance.platform, instance.instance_type, instance.public_ip_address, instance.image.id, instance.state))
             instance.stop()
-            if verbose:
-                logger.info('After stop function:\n')
-                logger.info("Id: {0}, Platform: {1}, Type: {2}, Public IPv4: {3}, AMI: {4}, State: {5}\n".format(instance.id, instance.platform, instance.instance_type, instance.public_ip_address, instance.image.id, instance.state))
+            logger.info("Stop the instance - Id: {0}, Type: {1}, Public IPv4: {2}, State: {3}".format(instance.id, instance.instance_type, instance.public_ip_address, instance.state))
 
-def stop_machines(done_path, type_machine, verbose, argv):
-    print('Stop machines...')
-    ec2 = boto3.resource('ec2', region_name='us-west-2')
-    list_instances = ec2.instances.all()  
-    types=type_machine.split('_')
-    for instance in list_instances:
-        if not instance.state['Name'] == 'stopped' and  instance.instance_type in types:
-            stop(instance, verbose)
-
-    with open(done_path, 'w') as f:
-            f.write(' '.join(argv) + '\n')    
+def stop_machines(type_machine, name_machines, logger):
+    logger.info('Stop machines...')
+    try:
+        ec2 = boto3.resource('ec2', region_name='us-west-2')
+    except:
+        logger.info(f'{datetime.datetime.now()}: Can\'t connect to ec2 with boto3 for stop the machines')
+        return
 
-if __name__ == "__main__":    
-    print(f'Starting {sys.argv[0]}. Executed command is:\n{" ".join(sys.argv)}')
-    parser = argparse.ArgumentParser()
-    parser.add_argument('done_path', type=str, help='A path to file that signals that the script finished running successfully.')
-    parser.add_argument('--type_machines', default='t2.medium_t2.2xlarge_m5a.24xlarge', help='stop only parts of types machines')
-    parser.add_argument('-v', '--verbose', action='store_true', help='Increase output verbosity')
+    list_instances = ec2.instances.all()
 
-    args = parser.parse_args()
-    if args.verbose:
-        logging.basicConfig(level=logging.DEBUG)
-    else:
-        logging.basicConfig(level=logging.WARNING)
-    logger = logging.getLogger('main')
-
-    stop_machines(args.done_path, args.type_machines, args.verbose, sys.argv)
+    types =type_machine.split('_') if type_machine else '' 
+    names =name_machines.split('_') if name_machines else ''
+    
+    for instance in list_instances:
+        if not instance.state['Name'] == 'stopped' and (types[0]=='all' or instance.instance_type in types) and (names[0]=='all' or instance.tags[0]['Value'] in names):
```

Comment:
instead of all check if empty.
Name should be pattern, no exact name, e.g., stop all workers using "worker*".
Also, if machine requires both instance type and name matching then specify it in command argument definition or change condition to or